### DISCUSSION TITLE
Errors Running bun run web and bun run dev:web

### DISCUSSION THREAD

**Discussion by maxbodin** [expo-starter/expo-template#28] [Q&A] [answered]
# Description:

I'm encountering errors when trying to run my Expo project for the web using Bun as the bundler. The following two commands lead to unresolved module errors:

# Steps to Reproduce:

Run : bunx create-expo-app --template @expo-starter/template

Run the project for the web using bun run web or bun run dev:web.

# Expected Behavior:
The app should bundle successfully for the web, and I should be able to view it in the browser.

# Actual Behavior:
The bundling process fails with the following errors.

# Screenshots:
1. bun run web output:

<img width="973" alt=" bun run web" src="https://github.com/user-attachments/assets/442c67f5-f1d7-48ff-a462-5bbd373f26d0">

2. bun run dev:web output:

<img width="1902" alt="bun run dev web" src="https://github.com/user-attachments/assets/6deaaba7-2152-4bf8-9ca7-56e9d3420709">

Please let me know if more information is needed. Thank you!

**Comment by AchrafBn** [comment]
Hello, can you provide  the output of the command `bunx expo-env-info` ?

**Reply by maxbodin** [reply]
Hello, thank you for your reply. Here is the output of the command :

<img width="683" alt="image" src="https://github.com/user-attachments/assets/2ca30509-b62a-41b4-8439-adfdcfcb4ea8">

**Reply by AchrafBn** [reply]
@maxbodin I faces the same error when switching from Node v18 to v22 but the app still bundles successfully, but I believe the issue appears because Expo Router supports Node v18, which is currently the stable version. Try downgrading with` nvm use 18` and see if the app bundles correctly

**Reply by maxbodin** [reply]
I downgraded to Node v18, but I'm still encountering the same issue.

<img width="988" alt="dev run web node18" src="https://github.com/user-attachments/assets/29a4c8a0-249f-47c0-8dfa-bbd590c3a7c8">

**Reply by AchrafBn** [reply]
@maxbodin @maxbodin  try re-run the command `bunx create-expo-app --template @expo-starter/template`

**Reply by maxbodin** [reply]
Still not working, I am having the following error when running bun run web : 

Web Bundling failed 5658ms index.ts (3103 modules)
error: ReferenceError: SHA-1 for file TestExpo/my-app/node_modules/.cache/nativewind/global.css.web.css (TestExpo/my-app/node_modules/.cache/nativewind/global.css.web.css) is not computed.

**Reply by maxbodin** [reply, marked as answer]
Running the following command resolved the issue for me. It seems that the problem originates from Tailwind:

```bash
yarn tailwindcss -i ./app/global.css -o ./node_modules/.cache/nativewind/global.css.web.css
```

From : https://github.com/nativewind/nativewind/issues/593#issuecomment-2355681096

**Comment by matisbyar** [comment]
Hi, I have the same issue! I did the same things as @maxbodin, which led me to same conclusions.
Thanks. 
MB

<img width="682" alt="Screenshot 2024-09-30 at 3 54 38 PM" src="https://github.com/user-attachments/assets/5b2c4fb4-2129-4544-b021-6cae7507e3ca">
